[ChinJunAn] iP

ChinJunAn opened the pull request:
### DukePro

> “We suffer more often in imagination than in reality.” - Seneca  [(Source)](https://www.tikvahlake.com/blog/essential-stoic-quotes-for-recovery/)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. 1. double-click it.
3. 2. add your tasks.
4. 3. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main `method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ChinJunAn committed:
Program can gather user input and echo back. Input stops when user enters "bye".

ChinJunAn committed:
Added the ability to store user input into an arraylist. If the input is "list", all items in the list will be displayed. Exits when user input is "bye".

ChinJunAn committed:
Improved list function. User can now mark and unmark task

ChinJunAn committed:
Added ToDo, Event, and Deadline tasks. Revamped function of storing list of tasks.

ChinJunAn committed:
Automated testing for "ToDo" function

ChinJunAn committed:
Added exception handling for wrong inputs.

ChinJunAn committed:
Improved encapsulation. Added delete function.

ChinJunAn committed:
Comments for future change. Improved coding conventions

ChinJunAn committed:
Add sava functionality. Save location in text file: "task.txt"

ChinJunAn committed:
Add date and time reading functionalities. Group "Deadlines" and "Events" classes under "TimedTask" to allow easier manipulation of date and time reading functionalities

ChinJunAn committed:
Include draft for search by date/time function in list command

ChinJunAn committed:
Merge branch 'branch-Level-7'

ChinJunAn committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadlines.java
#	src/main/java/Functions.java

ChinJunAn committed:
Improve OOP and exception handling.

ChinJunAn committed:
Organise into packages.

ChinJunAn committed:
Merge branch 'add-gradle-support' of https://github.com/ChinJunAn/ip

ChinJunAn committed:
Add 3 JUnit tests. Testing for:

- UI switch case
- Return statues of task
- Saving capabilities

ChinJunAn committed:
Correct path to Main method

ChinJunAn committed:
Improve save and load functionality. Subsequent run of duke program can remember previously saved tasks.

ChinJunAn committed:
Redo date time input and output formats.

ChinJunAn committed:
Add JavaDocs.

ChinJunAn committed:
Reformat according to java coding styles.

ChinJunAn committed:
Add find function to search by task description with a keyword.

ChinJunAn committed:
Merge branch 'branch-A-CodingStandard'

ChinJunAn committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Deadlines.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Events.java
#	src/main/java/duke/Functions.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/TimedTask.java
#	src/main/java/duke/ToDos.java

ChinJunAn committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Deadlines.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Events.java
#	src/main/java/duke/Functions.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/TimedTask.java
#	src/main/java/duke/ToDos.java
#	src/main/java/duke/UI.java

ChinJunAn committed:
Correct switch case style

ChinJunAn committed:
Correct java styles and add JavaDocs in UI class

ChinJunAn committed:
Peak version of GUI

ChinJunAn committed:
Merge branch 'Better-GUI' into branch-Level-10

ChinJunAn committed:
Add GUI. WIP

ChinJunAn committed:
Organise more packages and class, and improve class names

ChinJunAn committed:
Add input validation and adjust UI.

ChinJunAn committed:
Merge branch 'branch-Level-10'

Statspadders commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import duke.functions.Functions;
+import duke.gui.Window;
+import duke.storage.Storage;
+import duke.storage.TaskList;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.layout.Pane;
+import javafx.stage.Stage;
```
Clear import statements without using the asterisk * 👍

Statspadders commented on src/main/java/duke/buttons/DeadlineButton.java:
```diff
@@ -0,0 +1,38 @@
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.VBox;
+
+
```
Consider adding header class comments to public classes and methods

Statspadders commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @return Boolean to represent status of task. True for done, false for not done.
+     */
+    public boolean getStatus() {
+        return status;
+    }
```
Consider using the method isStatus() instead of getStatus() to follow the coding convention

Statspadders commented on src/main/java/duke/buttons/EventButton.java:
```diff
@@ -0,0 +1,41 @@
+        Label endLabel = new Label("End (yyyy-mm-dd hh:mm):");
+        TextField endTextField = new TextField();
+        Button addTaskButton = new Button("Add Task");
+        vbox.getChildren().addAll(desLabel, desTextField, startLabel, startTextField, endLabel, endTextField, addTaskButton);
```
Consider going to a new line as the code sentence is getting long

Statspadders reviewed (commented):
LGTM, almost. Just a few nits to fix. 
Overall, very clear and consistent coding style. Good job ! 👍

KevinEyo1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * Represents a Duke program.
+     */
+    private Functions fn;
```
Perhaps the naming could be more clear?

KevinEyo1 commented on src/main/java/duke/functions/CreateTodo.java:
```diff
@@ -0,0 +1,28 @@
+     *
+     * @param inp Description of todo task
+     */
+    public void todo(String inp) {
```
Perhaps the naming of the method could be clearer to show its usage?

KevinEyo1 commented on src/main/java/duke/functions/FindTask.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    public void find(String query) {
+        boolean flag = false;
+        boolean printed = false;
```
Perhaps these booleans could be renamed isFlag and isPrinted to follow naming convention?

KevinEyo1 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,92 @@
+     * Represent Event tasks. Event tasks have both start date time and end date time.
+     */
+    LocalDateTime start;
+    String consoleStartString;
```
I like that the naming is detailed enough and also follows camel case.

KevinEyo1 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,87 @@
+    /**
+     * Represent a general Task.
+     */
+    boolean status;
```
Could this boolean be represented as a question?

KevinEyo1 reviewed (commented):
LGTM, just a few nits to fix.

ChinJunAn committed:
Add assertions

Added assertions in TimedTask and Event class to check if conversion
of String to LocalDateTime object has occurred appropriately.

Correct JavaDocs

Corrected JavaDocs of function definitions in corresponding button
classes to reflect the appropriate parameters

ChinJunAn committed:
Window makeWindow: extract ScrollPane initialization

makeWindow has codes for initializing ScrollPane.

Code quality can be improved by extracting that body of code out
to a method.

Let's,
* extract the associated codes for initializing a ScrollPane into
another method

Ensure line wrappings follow Java code styles

ChinJunAn committed:
Merge pull request #1 from ChinJunAn/branch-A-Assertions

Branch a assertions

ChinJunAn committed:
Merge branch 'master' of https://github.com/ChinJunAn/ip

* 'master' of https://github.com/ChinJunAn/ip:
  Window makeWindow: extract ScrollPane initialization
  Add assertions

ChinJunAn committed:
Add sort function

TaskList could be jumbled with ToDo task that may not be as
important as TimedTask.

Made list command have sort function to allow list to display
closest deadline as the first item in listed Tasks.

Remove association between all commands and Function class

Must create a new instance whenever command is called. This uses
up more memory, and may be slower.

Make all commands static methods, with Function as one of its
parameter

ChinJunAn committed:
Merge pull request #2 from ChinJunAn/branch-C-Sort

Add sort function

ChinJunAn committed:
Merge branch 'master' of https://github.com/ChinJunAn/ip

* 'master' of https://github.com/ChinJunAn/ip:
  Add sort function

ChinJunAn committed:
Add files via upload

ChinJunAn committed:
Update README.md

ChinJunAn committed:
Correct mistakes from email and added headers for methods in
button package

ChinJunAn committed:
Merge branch 'master' of https://github.com/ChinJunAn/ip

ChinJunAn committed:
Add date picker, drop down menu for selecting time, task index for
marking/unmarking and deleting. Improve code quality.

Use text fields to gather user input.

This increases the chances for exceptions due to wrong user input.

Reduce user input in text fields to only specifying task description and
search function. Add drop down menus and date time picker to reduce user
input errors.

This reduces the chance for errors and wrong user input. Added benefit
for handling lesser exceptions due to wrong user input. Also, users
would not have to worry about date time format.

ChinJunAn committed:
Handle all exceptions and error handling, improve code quality, add
personality.

All known exceptions will not be printed onto console anymore.

Exceptions were printed onto the console and the user would not know
what actions led to an error.

Handle all exceptions and print reason for particular error onto
output pane in GUI.

This will make it known to the user that they have performed an action
that is not appropriate.

ChinJunAn committed:
Merge branch 'branch-A-CodeQuality'

ChinJunAn committed:
Update README.md

ChinJunAn committed:
Delete Ui.PNG

ChinJunAn committed:
Add files via upload

ChinJunAn committed:
Update README.md

ChinJunAn committed:
Delete Ui.PNG

ChinJunAn committed:
Add files via upload

ChinJunAn committed:
Delete Ui.PNG

ChinJunAn committed:
Add files via upload
